File: 12. Bitplane Slicing/Bitplane.m
img = imread("bitplane.tif");
i1 = bitget(img, 1);
i2 = bitget(img, 2);
i3 = bitget(img, 3);
i4 = bitget(img, 4);
i5 = bitget(img, 5);
i6 = bitget(img, 6);
i7 = bitget(img, 7);
i8 = bitget(img, 8);
imshow(i1);
imshow(i1*255);
imshow(i5*255);
imshow(i6*255);
imshow(i7*255);
imshow(i8*255);
new = i8*128 + i7*64 + i6*32;
imshow(img);
figure;
imshow(new);

File: 12. Bitplane Slicing/Bitplane.py
import cv2
import numpy as np
from matplotlib import pyplot as plt

img = cv2.imread('bitplane.tif', 0)
r, c = img.shape
recon = np.full((r, c), 0, np.uint8)
bit = list(range(0, 8))
for k in bit:
    plane = np.full((r, c), 2**k, np.uint8)
    res = cv2.bitwise_and(plane, img)
    new_img = res*255
    cv2.imwrite("bitplane" + str(k) + " .png", new_img)
    if k in [5, 6, 7]:
        recon = cv2.bitwise_or(recon, res)
    cv2.imwrite("bit_recon.png", recon)
